state_machine/state_engine.py

```
@@ -120,6 +125,23 @@ def set_phase(self, phase: MissionPhase) -> Dict[str, Any]:
+        with self._transition_lock:
+            # Check if another transition is already in progress
+            if self._is_transitioning:
+                raise StateTransitionError(
+                    "Phase transition already in progress. Please wait for the current transition to complete.",
+                    component="state_machine",
+                    context={
+                        "current_phase": self.current_phase.value,
+                        "target_phase": str(phase),
+                        "status": "transition_in_progress"
+                    }
+                )
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)